src/Apps/Order/Routes/Shipping2/ShippingContext.ts

```diff
@@ -1,20 +1,342 @@
-import { FulfillmentValues } from "Apps/Order/Routes/Shipping2/FulfillmentDetails"
-import { createContext, useContext } from "react"
+import {
+  createContext,
+  useCallback,
+  useContext,
+  useMemo,
+  useState,
+} from "react"
+import { FormikProps } from "formik"
+import { ShippingProps, ShippingRouteStep } from "Apps/Order/Routes/Shipping2"
+import {
+  FulfillmentValues,
+  ShipValues,
+} from "Apps/Order/Routes/Shipping2/FulfillmentDetails"
+import {
+  addressWithFallbackValues,
+  getDefaultUserAddress,
+  FulfillmentType,
+  ShippingAddressFormValues,
+} from "Apps/Order/Routes/Shipping2/shippingUtils"
+import { SavedAddressType } from "Apps/Order/Utils/shippingUtils"
+import { extractNodes } from "Utils/extractNodes"
+import { ALL_COUNTRY_CODES, EU_COUNTRY_CODES } from "Components/CountrySelect"
+import { COUNTRIES_IN_EUROPEAN_UNION } from "@artsy/commerce_helpers"
+// TODO: Duplicated list ^
 
+type FulfillmentHelpers = Pick<
+  FormikProps<FulfillmentValues>,
+  "handleSubmit" | "isValid" | "values"
+>
 export interface ShippingContextProps {
-  lockShippingCountryTo: "EU" | string | null
-  shipsFrom: string
-  availableShippingCountries: string[]
-  requiresArtsyShippingTo: (shipTo: string) => boolean
-  selectedSavedAddressId: string | null
-  fulfillmentDetails: FulfillmentValues["attributes"] | null
-  fulfillmentType: FulfillmentValues["fulfillmentType"] | null
-  isArtsyShipping?: boolean
-  shippingQuotes?: Array<{ id: string; isSelected: boolean }>
+  savedOrderData: {
+    lockShippingCountryTo: "EU" | string | null
+    shipsFrom: string
+    availableShippingCountries: string[]
+    requiresArtsyShippingTo: (shipTo: string) => boolean
+    selectedSavedAddressId: string | null
+    fulfillmentDetails: FulfillmentValues["attributes"] | null
+    fulfillmentType: FulfillmentValues["fulfillmentType"] | null
+    isArtsyShipping?: boolean
+    shippingQuotes?: Array<{ id: string; isSelected: boolean }>
+  }
+  initialValues: {
+    fulfillmentDetails: FulfillmentValues
+    shippingQuotes: {
+      selectedShippingQuoteId: any
+    }
+  }
+  step: ShippingRouteStep
+  helpers: {
+    fulfillmentDetails: FulfillmentHelpers & {
+      setFulfillmentFormHelpers: (helpers: FulfillmentHelpers) => void
+    }
+  }
 }
```

Comment:
Major change here is to the context, which is now much less stringent about what it includes (it includes a lot more, and other types are simpler). Thanks to this i've moved all context loading stuff from index.tsx to this file. I wonder if it is worth adding comments to give us more insight in our vscode type definitions on hover for some of them.